resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/kubernetes/KubernetesClusterSchedulerBackend.scala

```diff
@@ -218,6 +245,20 @@ private[spark] class KubernetesClusterSchedulerBackend(
 
   }
 
+  private def updateJobResourceStatus(key: String, value: Any): Unit = {
+    logInfo(s"Updating Job Resource with name. $jobResourceName")
+    sparkJobResourceController.foreach(controller =>
+      Try(
```

Comment:
Yep, and that might eliminate this method entirely by moving everything into the `updateJobObject` call